flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/Table.java

```diff
@@ -879,4 +879,117 @@
 	 * @return An OverWindowedTable to specify the aggregations.
 	 */
 	OverWindowedTable window(OverWindow... overWindows);
+
+	/**
+	 * Performs a field add operation. Similar to an SQL SELECT statement. The field expressions
+	 * can contain complex expressions, but can not be an aggregations. Will report an expression if
+	 * the added field name already exists.
+	 *
+	 * <p>Example:
+	 * <pre>
+	 * {@code
+	 *   tab.addColumns("a + 1 as a1, concat(b, 'sunny') as b1")
+	 * }
+	 * </pre>
+	 */
+	Table addColumns(String fields);
+
+	/**
+	 * Performs a field add operation. Similar to an SQL SELECT statement. The field expressions
+	 * can contain complex expressions, but can not be an aggregations. Will report an expression if
+	 * the added field name already exists.
+	 *
+	 * <p>Scala Example:
+	 *
+	 * <pre>
+	 * {@code
+	 *   tab.addColumns('a + 1 as 'a1, concat('b, "sunny") as 'b1)
+	 * }
+	 * </pre>
+	 */
+	Table addColumns(Expression... fields);
+
+	/**
+	 * Performs a field add operation. Similar to an SQL SELECT statement. The field expressions
+	 * can contain complex expressions, but can not be an aggregations. Existing fields will be
+	 * replaced if {@code replaceIfExist} is true.
+	 *
+	 * <p>Example:
+	 * <pre>
+	 * {@code
+	 *   tab.addColumns(true, "a + 1 as a1, concat(b, 'sunny') as b1")
+	 * }
+	 * </pre>
+	 */
+	Table addColumns(boolean replaceIfExist, String fields);
+
+	/**
+	 * Performs a field add operation. Similar to an SQL SELECT statement. The field expressions
+	 * can contain complex expressions, but can not be an aggregations.  Existing fields will be
+	 * replaced if {@code replaceIfExist} is true.
+	 *
+	 * <p>Scala Example:
+	 * <pre>
+	 * {@code
+	 *   tab.addColumns(true, 'a + 1 as 'a1, concat('b, "sunny") as 'b1)
+	 * }
+	 * </pre>
+	 */
+	Table addColumns(boolean replaceIfExist, Expression... fields);
+
+	/**
+	 * Performs a field rename operation. Similar to an field alias statement. The field expressions
+	 * should be an alias expressions, and only the existing fields can be renamed.
+	 *
+	 * <p>Example:
+	 *
+	 * <pre>
+	 * {@code
+	 *   tab.renameColumns("a as a1, b as b1")
+	 * }
+	 * </pre>
+	 */
+	Table renameColumns(String fields);
+
+	/**
+	 * Performs a field rename operation. Similar to an field alias statement. The field expressions
+	 * should be an alias expressions, and only the existing fields can be renamed.
+	 *
+	 * <p>Scala Example:
+	 *
+	 * <pre>
+	 * {@code
+	 *   tab.renameColumns('a as 'a1, 'b as 'b2)
+	 * }
+	 * </pre>
+	 */
+	Table renameColumns(Expression... fields);
+
+	/**
+	 * Performs a field drop operation. The field expressions
+	 * should be an field reference expressions, and only the existing fields can be drop.
```

Comment:
"an" should be removed
only the existing fields can be drop -> only existing fields can be dropped